packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -121,6 +140,57 @@ contract FuelERC20GatewayV4 is
         require(success);
     }
 
+    /**
+     * @notice Intitializing rate limit params for each token. 
+     * @param _token The token address to set rate limit params for.
+     * @param _limitAmount The amount to reset the limit to.
+     * @param _rateLimitDuration rate limit duration.
+     */
+    function initializeRateLimit(address _token, uint256 _limitAmount, uint256 _rateLimitDuration) external onlyRole(SET_RATE_LIMITER_ROLE) {
+        if (currentPeriodEnd[_token] != 0) revert RateLimitAlreadySet();
```

Comment:
Why isn't it possible to change the duration after it has been set?

Reply:
so the workflow would be to call initialzeRateLimit() first then to change the config `resetRateLimit()` must be used?

also good to emit an event here. can be generic as `RateLimitUpdated()` which can be used in both methods.

Reply:
Please make it dynamic. The reasons why it is static in FuelMessagePortal is that we can change it via new deployment (it is a constant / immutable), but here we cannot leverage that advantage, as we need an storage variable for each token. Thus it makes no sense to have that limitation.